guides/tutorials/codelabs/pubsub-to-appengine/index.md

```diff
@@ -54,17 +61,23 @@ If Spinnaker is not yet configured, follow this Halyard [quickstart](https://www
 
 ### Configure Spinnaker to listen to the Google Cloud Pub/Sub subscription
 
-Configure your GCS artifact provider.
+First, configure your GCS artifact provider.
+
+1. Enable artifiact support:   
+`hal config features edit --artifacts true`.
 
-1. `hal config features edit --artifacts true`.
+2. Enable the GCS artifact provider:   
+`hal config artifact gcs enable`
 
-2. `hal config artifact gcs enable`.
+3. Add Spinnaker configuration for the GCS artifact provider's account:   
+`hal config artifact gcs account add --json-path <json_service_key> <artifact_account>`
 
-3. `hal config artifact gcs account add --json-path <key> <artifact_account>`.
+The json_service_key can be one returned after you configured your GAE cloud provider, above, or one you got when you configured your GCS storage (ppossibly in the [Halyard on GCE Quickstart](https://www.spinnaker.io/setup/quickstart/halyard-gce/))
```

Comment:
s/ppossibly/possibly. 

It's also important to note here that the service account associated with that key needs to have GCS write capabilities, no matter where the key originates from (you could even create a new key just for this).